src/V3LinkParse.cpp

```diff
@@ -247,7 +247,10 @@ class LinkParseVisitor final : public VNVisitor {
         // Maybe this variable has a signal attribute
         V3Config::applyVarAttr(m_modp, m_ftaskp, nodep);
 
-        if (v3Global.opt.publicFlatRW()) {
+        if (v3Global.opt.publicFlatRW()
+            || (v3Global.opt.publicDepth() && m_modp
+                && (m_modp->level() - 1) <= v3Global.opt.publicDepth())) {
```

Comment:
If a module is used once at a high level and once much lower I don't think this will do what the user expects, which is to enable it.  I think V3LinkCells would need to do the appropriate minimum level checking and so implement this.

Reply:
Would the AstVars from these different hierarchies be different instances? It looks like v3EmitSyms reads this value, not anything on AstCell, which looks like what v3LinkCells works with. Should I add an AstVar visitor to v3LinkCells? Or is there some visitor that will link to an AstVar or a list of them?

Reply:
Doing this seems similar to https://github.com/verilator/verilator/issues/3950. Would I have to do something to make sure a scope is created? Also not quite sure what that means.

Reply:
A scope is the hierarchy for one specific instance of a cell of a module.  For example you might have a.x and a.b.c.x where x are both cells of module X, and a.b.c.y where y is of module Y.  I'd expect publicDepth 2 to in this case ideally do just a.x (but as we described earlier that's hard), so it see a.x and a.b.c.x and so make X public and thus a.x and a.b.c.x public. But would not make a.b.c.y and therefore Y public.

As I said above "a.x, a.b.c.x" would have rank 4.  In GraphAlgRank it is passing the rank down and doing a <=. Instead if you add a flag to this algorithm to invert it to >= (that is compute shortest distance).  Then use that (perhaps after saving in the AstNodeModule) to compare against your distance.

BTW do you expect to use public for more than the top level with this flag? That's much easier.

Reply:
Yeah I do mainly want to just use this for the top level, so I'm honestly fine with the current implementation.

Reply:
Ok, then just add to docs that the behavior may be unpredictable for multiply instantiated modules with values greater than 1.